text/0177-mintless-jetton-standard.md

```diff
@@ -0,0 +1,90 @@
+- **TEP**: [177](https://github.com/ton-blockchain/TEPs/pull/177)
+- **title**: Mintless jettons
+- **status**: Draft
+- **type**: Contract Interface
+- **authors**: [Emelyanenko Kirill](https://github.com/EmelyanenkoK),  [Denis Subbotin](https://github.com/mr-tron)
+- **created**: 7.07.2024
+- **replaces**: -
+- **replaced by**: -
+
+# Summary
+
+Extension for [Jetton](https://github.com/ton-blockchain/TEPs/blob/master/text/0074-jettons-standard.md) that allows merkle-proof airdrops with mint occuring on jetton-wallet contract in decentralised manner.
+Support from wallet apps will allow to show not yet claimed jettons on balance and automatically claim on first outgoing transfer.
+
+# Motivation
+
+There is a need (to create large communities, big advertising campaigns, etc) for a type of Jettons that can be simultaneously airdropped to millions of users and being deployed on demand without large costs and significant additional load to blockchain.
+
+# Guide
+
+A straightforward way to achieve properties described in [Motivation](#motivation) is using [Merkle trees](https://en.wikipedia.org/wiki/Merkle_tree) - when airdrop amount for each user is stored in a leaf of the tree and can be "claimed" by owner by providing a small proof of its presence in the tree, and the jetton itself only needs to store a single hash for all items (not per item).
+
+### Useful links
+
+1. [Reference contract implementation](https://github.com/ton-community/mintless-jetton)
+2. [Custom payload API](ttps://github.com/ton-blockchain/TEPs/blob/master/text/0000-jetton-offchain-payloads.md)
```

Comment:
**h**ttps, missing "h" at the beginning

Reply:
EQCDjUcZdVCmk7myvIuZqJ7WGC1UW0845XE2rmCPKkGWiVDO